File: benchmark/tests/threads-single/threads_001.lua

function process(ini, fim)
	local acc = 0;
	local i = ini;

	while (i <= fim) do
		acc = acc + i;
		i = i+1;
	end

	return acc;
end

local L = 10000000;
print("Lua");
print(string.format('%d',process(0, 2 * L)));

File: benchmark/tests/threads-single/threads_001.py
from multiprocessing import Process, Queue

def process(r, ini, fim):
	acc = 0
	i = ini

	while (i <= fim):
		acc = acc + i
		i = i+1

	r.put(acc)


if __name__ == '__main__':
	L = 10000000
	n = 2 * L
	acc = Queue()
	process(acc, 0, n)
	print 'python'
	print acc.get()
